Content.Client/Gameplay/GameplayState.cs

```diff
@@ -26,73 +36,113 @@ public sealed class GameplayState : GameplayStateBase, IMainViewportState
         [Dependency] private readonly IConfigurationManager _configurationManager = default!;
         [Dependency] private readonly IEntityManager _entMan = default!;
 
-        protected override Type? LinkedScreenType => typeof(DefaultGameScreen);
-        public static readonly Vector2i ViewportSize = (EyeManager.PixelsPerMeter * 21, EyeManager.PixelsPerMeter * 15);
         private FpsCounter _fpsCounter = default!;
 
-        public MainViewport Viewport { get; private set; } = default!;
+        public MainViewport Viewport => _uiManager.ActiveScreen!.GetWidget<MainViewport>()!;
+
+        private readonly GhostUIController _ghostController;
+        private readonly ActionUIController _actionController;
+        private readonly AlertsUIController _alertsController;
+        private readonly HotbarUIController _hotbarController;
+        private readonly ChatUIController _chatController;
+        private readonly ViewportUIController _viewportController;
+        private readonly GameTopMenuBarUIController _menuController;
 
         public GameplayState()
         {
             IoCManager.InjectDependencies(this);
+
+            _ghostController = _uiManager.GetUIController<GhostUIController>();
+            _actionController = _uiManager.GetUIController<ActionUIController>();
+            _alertsController = _uiManager.GetUIController<AlertsUIController>();
+            _hotbarController = _uiManager.GetUIController<HotbarUIController>();
+            _chatController = _uiManager.GetUIController<ChatUIController>();
+            _viewportController = _uiManager.GetUIController<ViewportUIController>();
+            _menuController = _uiManager.GetUIController<GameTopMenuBarUIController>();
```

Comment:
cant you `[Dependency]` these or does that only work in other UI controllers

Reply:
only works in other UI controllers